llvm_mode/compare-transform-pass.so.cc

```diff
@@ -341,31 +348,20 @@ bool CompareTransform::transformCmps(Module &M, const bool processStrcmp,
     Value *     VarStr;
     bool        HasStr1 = getConstantStringInfo(Str1P, Str1);
     bool        HasStr2 = getConstantStringInfo(Str2P, Str2);
-    uint64_t    constLen, sizedLen;
+    uint64_t    constStrLen, constSizedLen, unrollLen;
     bool        isMemcmp =
         !callInst->getCalledFunction()->getName().compare(StringRef("memcmp"));
     bool isSizedcmp = isMemcmp ||
                       !callInst->getCalledFunction()->getName().compare(
                           StringRef("strncmp")) ||
                       !callInst->getCalledFunction()->getName().compare(
                           StringRef("strncasecmp"));
+    bool isConstSized = isSizedcmp && isa<ConstantInt>(callInst->getArgOperand(2));
```

Comment:
What about getArgOperand(1)? *cmp() function's first argument could be constSized, right?

Reply:
constSize means: memcmp(foo, bar, 5). if there is a '2' it is a constant len. in contrast to memcmp(foo, bar, len) where it is a variable that can not be resolved. operand(0) is foo, operand(1) is bar, operand(2) is the length parameter, either a const value or a variable. its fine.

Reply:
Oops, i mixed up the arg numbering, sorry. Thanks for clarification.